c++/src/kj/list.h

```diff
@@ -202,6 +203,231 @@ class ListIterator {
   friend class List<T, link>;
 };
 
-} // namespace kj
+template <typename T>
+class OwnList {
+  // A linked list that does its own memory allocation.
+  //
+  // Keep in mind that each new element requires the allocation of two pointers and a T. This is
+  // very little if T is actually an Own<U> but can be quite expensive if T is a large type.
+
+  struct Node {
+    // A simple linked list node which extends the lifetime of the next node.
+
+    Node(T&& value): value{ kj::mv(value) } {}
+
+    Maybe<Own<Node>> maybeNext;
+    Node* prev = nullptr;
```

Comment:
I'm familiar with this variant of linked lists. I appreciate that it decreases the need for branching logic. In some situations, it may lead what was previously a single deref into a double deref, which is less of an issue since we use tcmalloc. I do worry that it makes code more difficult to follow for a casual reader. I've rewritten to a variant of this with a paranoid amount of comments.